serde/src/java/org/apache/hadoop/hive/serde2/io/TimestampWritableV2.java

```diff
@@ -374,16 +374,6 @@ public String toString() {
       return timestamp.toString();
     }
 
-    String timestampString = timestamp.toString();
-    if (timestampString.length() > 19) {
```

Comment:
The character length is at a maximum of 19, indicating that the current design supports a maximum time of: **9999-12-31 23:59:59**.

This is an interesting issue. If this were a real business use case, the year 9999 would absolutely be an unreachable limit.

So, for this PR, if users want to perform some testing based on impossible years (>9999), then this PR might still have some meaning. Otherwise, Apache Hive living until 9999 is already more than enough! Haha.

Reply:
:), still, why do we need this hack in a code?